ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/BlockTree.hs

```diff
@@ -76,7 +77,7 @@ mkTrunk btTrunk = BlockTree { btTrunk, btBranches = [] }
 -- block in common with an existingbranch.
 addBranch :: AF.HasHeader blk => AF.AnchoredFragment blk -> BlockTree blk -> Maybe (BlockTree blk)
 addBranch branch BlockTree{..} = do
-  (_, btbPrefix, _, btbSuffix) <- AF.intersect btTrunk branch
+  (_, btbPrefix, btbTrunkSuffix, btbSuffix) <- AF.intersect btTrunk branch
```

Comment:
Actually, shouldn't it be
```suggestion
  (btbPrefix, _, btbTrunkSuffix, btbSuffix) <- AF.intersect btTrunk branch
```

Reply:
The chain generator is a bit of a degenerated case where both fragments share the same anchor. However, the interface of `BlockTree` does not make such assumptions and, in the event that it is reused in the future in a different way (by us or others), I think this would make it more robust, considering that the documented pre-condition for `addBranch` is only that the fragment has to be anchored in the trunk, not necessarily at the trunk's anchor.